ui.apps/src/main/content/jcr_root/apps/venia/clientlibs/common/js/CommerceApi.js

```diff
@@ -0,0 +1,203 @@
+/*******************************************************************************
+ *
+ *      Copyright 2019 Adobe. All rights reserved.
+ *      This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License. You may obtain a copy
+ *      of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software distributed under
+ *      the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *      OF ANY KIND, either express or implied. See the License for the specific language
+ *      governing permissions and limitations under the License.
+ *
+ *
+ ******************************************************************************/
+window.CIF = window.CIF || {};
+
+/**
+ * The CommerceApi is responsible for interacting with the Commerce backend using REST API calls
+ */
+(function () {
+
+    'use strict';
+    const endpoints = {
+        guestcarts: {
+            create: '/guest-carts',
+            byId: (id) => (`/guest-carts/${id}`),
+            addEntry: (id) => (`/guest-carts/${id}/items`),
+            totals: (id) => (`/guest-carts/${id}/totals`),
+            itemOperation: (cartId, itemId) => (`/guest-carts/${cartId}/items/${itemId}`)
+        }
+
+    };
+
+    class CommerceApi {
+
+        /**
+         * initializes the commerce API.
+         * @param props {Object} the props have the following structure: { endpoint }
+         */
+        constructor(props) {
+            if (!props.endpoint) {
+                throw new Error('The commerce API is not properly initialized. The "endpoint" property is missing from the initialization object');
+            }
+
+            this.rootEndpoint = props.endpoint;
+        }
+
+        /**
+         * Issues a request to the supplied URL using the provided parameters
+         * @param url
+         * @param params
+         * @returns {Promise<any>} the JSON response or throws an error
+         * @private
+         */
+        async _fetch(url, params) {
+            let response = await fetch(url, params);
+            if (!response.ok) {
+                let message = await response.text();
+                throw new Error(message);
+            }
+            return response.json();
+        }
+
+        /**
+         * Performs an update operation (POST or PUT).
+         * @param url the request URL
```

Comment:
That parameter is called `endpoint` in the method, and shouldn't this be consistent with `_fetch`? You don't "build" the `url` at the same method level, I think it would be more consistent to do it everywhere at the same place.